helper-cli/src/main/kotlin/commands/SetDependencyRepresentationCommand.kt

```diff
@@ -0,0 +1,170 @@
+        .required()
+
+    private val outputFile by option(
+        "--output-file", "-o",
+        help = "The file in which to write the result of the conversion. If missing, the input file is overwritten."
+    ).convert { it.expandTilde() }
+        .file(mustExist = false, canBeFile = true, canBeDir = false, mustBeWritable = false, mustBeReadable = false)
+        .convert { it.absoluteFile.normalize() }
+
+    private val targetFormat by option(
+        "--format", "-f",
+        help = "The target format for the conversion."
+    ).enum<TargetFormat>().default(TargetFormat.GRAPH)
+
+    override fun run() {
+        val converters = listOf(::convertOrtResult, ::convertAnalyzerResult, ::convertProjectAnalyzerResult)
+        if (converters.none { it() }) {
+            throw UsageError("No supported data type could be found in the input file.")
+        }
+    }
+
+    /**
+     * Convert a full ORT result file.
+     */
+    private fun convertOrtResult(): Boolean =
+        convertResult<OrtResult> { result ->
+            println("Converting a full ORT result file.")
+
+            val analyzerResult = convertToTarget(result.analyzer?.result)
+            val convertedResult = result.copy(
+                analyzer = result.analyzer?.copy(result = analyzerResult)
+            )
+
+            writeResult(convertedResult)
+        }
+
+    /**
+     * Convert an [AnalyzerResult] file.
+     */
+    private fun convertAnalyzerResult(): Boolean =
+        convertResult<AnalyzerResult> {
+            println("Converting an analyzer result file.")
+
+            writeResult(convertToTarget(it))
+        }
+
+    /**
+     * Convert a [ProjectAnalyzerResult]. In this case, the result file contains an [AnalyzerResult], as a
+     * [ProjectAnalyzerResult] does not contain a dependency graph.
+     */
+    private fun convertProjectAnalyzerResult(): Boolean =
+        convertResult<ProjectAnalyzerResult> { result ->
+            println("Converting a project analyzer result file.")
+
+            val analyzerResult = AnalyzerResultBuilder().addResult(result).build()
+            writeResult(convertToTarget(analyzerResult))
+        }
+
+    /**
+     * Convert a result of type [T]. Read this result from the input file and invoke [block] with it. Return *false*
+     * if the result file could not be loaded (which probably means that it contains a result of a different type).
+     */
+    private inline fun <reified T : Any> convertResult(block: (T) -> Unit): Boolean =
+        runCatching {
+            block(ortFile.readValue())
+        }.isSuccess
+
+    /**
+     * Convert the given [analyzerResult] to the selected target format. Fail if no result is available or the
+     * result is already in the target format.
+     */
+    private fun convertToTarget(analyzerResult: AnalyzerResult?): AnalyzerResult {
+        val convertedResult = targetFormat.convert(
+            analyzerResult ?: throw UsageError("No analyzer result available.")
```

Comment:
I believe the command is easier to use if you change semantics to let's say "the command converts all analyzer results contained in the ORT result to the specified format". This way the caller does not need to check first whether the ORT result contains an analyzer result. This makes it easier to use from a script.

Reply:
I agree to the approach to make the command easier to use in scripts. So I changed (or basically removed) the error handling. It is now only visible from the command's output whether a conversion was done or not.

The only thing I am not sure is whether an error code should be returned if the input file could not be processed at all. (Currently it is not, but this would be rather easy to change.)

Reply:
If execution is not successful because the input file does not exist or cannot be deserialized IMO a non zero exit code would be fine.